internal/common/function/function.go

```diff
@@ -137,11 +139,46 @@ func GenerateVeleroBackupName(namespace, nabName string) string {
 	return veleroBackupName
 }
 
-// CheckVeleroBackupLabels return true if Velero Backup object has required Non Admin labels, false otherwise
-func CheckVeleroBackupLabels(labels map[string]string) bool {
-	// TODO also need to check for constant.OadpLabel label?
-	value, exists := labels[constant.ManagedByLabel]
-	return exists && value == constant.ManagedByLabelValue
+// CheckVeleroBackupMetadata return true if Velero Backup object has required Non Admin labels and annotations, false otherwise
+func CheckVeleroBackupMetadata(obj client.Object) bool {
+	labels := obj.GetLabels()
+	if !checkLabelValue(labels, constant.OadpLabel, constant.OadpLabelValue) {
+		return false
+	}
+	if !checkLabelValue(labels, constant.ManagedByLabel, constant.ManagedByLabelValue) {
+		return false
+	}
+
+	annotations := obj.GetAnnotations()
+	if !checkAnnotationValueIsValid(annotations, constant.NabOriginNamespaceAnnotation) {
+		return false
+	}
+	if !checkAnnotationValueIsValid(annotations, constant.NabOriginNameAnnotation) {
+		return false
+	}
+	// TODO what is a valid uuid?
```

Comment:
should I have a different validation here? (from what we talked today, this is field will be removed, right? To add one ourselves)

Reply:
I think we can use go https://github.com/google/uuid to generate UUID ourselves.

Reply:
yes or another option would be https://github.com/satori/go.uuid
But Google one seems like its more maintained.

Reply:
@shubham-pampattiwar the satori one was last updated 6 years ago with ~37K uses, so I think it's better stick to something more active. google/uuid was updated 2 weeks ago so it's much more active project with ~320K uses.

Reply:
yes certainly, google/uuid is the way to go.